app/assets/javascripts/ng-control/browser.js

```diff
@@ -323,47 +323,75 @@
       $scope.views.search.status = "searching";
       $scope.views.search.page = p;
 
       $http.post("/browser/search",$scope.views.search).
-        then(function(response) { 
+        then( function(response) { 
           $scope.views.search.status = "preparing";         
-          $scope.views.search.samples = aq.collect(response.data.samples,function(s) {
-            var sample = new Sample($http).from(s);
-            if ( aq.url_params().sid && sample.id === parseInt(aq.url_params().sid) ) {
-              sample.open = true;
-              sample.inventory = true;
-              sample.loading_inventory = true;
-              sample.get_inventory(function() {
-                sample.loading_inventory = false;            
-                sample.inventory = true;
-              });              
-            }
-            return sample;
-          });
+          $scope.views.search.samples = sample_inventory(response.data.samples);
           $scope.views.search.count = response.data.count;
           $scope.views.search.pages = aq.range(response.data.count / 30);
           $scope.views.search.status = "done";
         });
           
     }
 
+    // item_search function allows users to search for sample by Item ID
     $scope.item_search = function() {
       AQ.Item.find($scope.views.search.item_id).then( item => {
-        $scope.views.search.item = item;
-        $scope.views.search.item.modal = true;
-        $scope.views.search.item.new_location = $scope.views.search.item.location;
-        cookie();
-        AQ.update();
-      }).catch(() => alert("Could not find item with id " + $scope.views.search.item_id));
+        AQ.ObjectType.find(item.object_type_id).then( object_type => {
+          // Collection check
+          if (object_type.handler === 'collection') {
+            AQ.Collection.find_fast(item.id).then( collection => {
+              var item_list = collection.part_matrix.reduce(function (a, b) {
+                return a.concat(b);
+              });
+              $scope.views.search.status = "preparing";
+              $scope.views.search.samples = sample_inventory([item_list[0].sample]);
+              $scope.views.search.count = 1;
+              $scope.views.search.pages = aq.range(1 / 30);
+              $scope.views.search.status = "done";             
+            })
+            
+          } else {
+            $scope.views.search.status = "preparing";
+            AQ.Sample.find(item.sample_id).then( sample => {
+              $scope.views.search.samples = sample_inventory([sample]);
+              $scope.views.search.count = 1;
+              $scope.views.search.pages = aq.range(1 / 30);
+              $scope.views.search.status = "done";
+          })
+            .catch(() => { 
+              alert("Could not find sample with item id " + $scope.views.search.item_id);
+              $scope.views.search.status = "done";
+            });
+          }
+        })
```

Comment:
There should be `catch` handlers for each promise.  If an error occurs inappropriately, then the call that led to it shouldn't happen.